### PR TITLE
Publish button

### LINKED ISSUE #372: Add publish button to site pages
Just like #250 but for publishing instead of previewing

### PR DESCRIPTION
**Pull request by mbertrand**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [x] Mobile width screenshots
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes #372 

#### What's this PR do?
Adds a publish button next to the preview button

#### How should this be manually tested?
Follow the instructions for setting up github integration.
Create or go to any site and click the 'Publish' button.
Go to your github organization, a repo for your site should be there and there should be a very recent commit to your 'release' branch.

#### Screenshots (if appropriate)
Default style:

Screen Shot 2021-07-08 at 2 37 53 PM

On hover:

Screen Shot 2021-07-08 at 2 38 01 PM

Narrow width:

Screen Shot 2021-07-08 at 2 41 08 PM

### CODE REVIEW

File: static/js/pages/SitePage.tsx [hunk 1]
```diff
@@ -46,6 +49,18 @@ export default function SitePage(): JSX.Element | null {
...
 
+  const onPublish = async () => {
+    if (publishIsPending) {
+      return
+    }
+    const response = await publishWebsite()
+    if (!response) {
+      return
+    } else {
+      // TBD
```
Comment (noisecapella) [inline static/js/pages/SitePage.tsx:60]: What's to be determined here?

**Comment by noisecapella** [comment]
Functionality looks good. Do you know if there's a plan yet for UI feedback for the preview or publish buttons?

**Comment by mbertrand** [comment]
There is an issue that briefly mentions it, going to work on that in a separate PR: https://github.com/mitodl/ocw-studio/issues/370

**Comment by noisecapella** [comment]
@mbertrand I see, would this `TBD` be taken care of as part of #370? https://github.com/mitodl/ocw-studio/pull/374#discussion_r669736094

File: static/js/pages/SitePage.tsx (hunk 1 above)
Comment (mbertrand) [reply static/js/pages/SitePage.tsx:60]: What to show in the UI based on the API response (success or failure)

**Comment by mbertrand** [comment]
Yes

**Review by noisecapella — APPROVED** [review]

**Pull request merged**